plugin/editorconfig.vim

```diff
@@ -383,6 +383,11 @@ function! s:ApplyConfig(config) abort " Set the buffer options {{{1
         return
     endif
 
+    " Do not process netrw buffers
+    if &b:filetype == 'netrw'
```

Comment:
There are global and local options [[1]](https://vimhelp.org/eval.txt.html#%3Alet-%26). There are no 'buffer' options. This is a syntax error that yields

```
E113: Unknown option: b
```

Unfortunately, this error is hidden by the try-catch from the [call site](https://github.com/editorconfig/editorconfig-vim/blob/6e2b0920f5836aaf882b43ee52700d3f509cdd1d/plugin/editorconfig.vim#L303-L310).

@pmeinhardt Please have a look. Maybe a change to `&l` is sufficient? I could not reproduce your original problem and therefore don't know if this is the right fix.

Reply:
Filed #215 for better visibility.

Reply:
Thank you for the catch -- I opened #216 to fix this.

Reply:
Thanks @kaihowl ❤️ Unfortunately with the change from #216 the problem came back. 🤔

Reply:
That is indeed unfortunate.

I can even reproduce this with nerdtree instead of netrw. There already is [a condition](https://github.com/editorconfig/editorconfig-vim/blob/5b875ac1aeba22abce3c499c0d5e4f33558fdf2c/plugin/editorconfig.vim#L372-L374) that should take care of not messing non-editable buffers. Nerdtree for example hijacks the `FileExplorer` event group and instead [sets the necessary options](https://github.com/preservim/nerdtree/blob/fc85a6f07c2cd694be93496ffad75be126240068/plugin/NERD_tree.vim#L157-L158), including `buftype`, as part of a new set of autocommands. These are sequenced after the editorconfig autocommands. Therefore, the `buftype` appears as editable when editorconfig sees it. The final value is of course `nofile`.

I am not an expert on autocommands. Therefore, my proposal for a fix is rather hacky: https://github.com/editorconfig/editorconfig-vim/pull/217